### PR TITLE
feat: add a deposit script parsing function

### PR DESCRIPTION
**Pull request by djordon**
## Description

Closes https://github.com/stacks-network/sbtc/issues/327.

## Changes

- Add a function that can parse a deposit script and extract the relevant pieces of information.

## Testing information

The tests check that the parser function is an inverse of the `DepositScriptInputs::deposit_script` function.

### CODE REVIEW

File: common/src/deposits.rs [hunk 1]
```diff
@@ -96,16 +197,67 @@ mod tests {
...
+        let parsed_deposit = parse_deposit_script(&deposit_script).unwrap();
+
+        assert_eq!(deposit, parsed_deposit);
+    }
+
     /// Basic check that we can create an address without any issues
     #[test_case(PrincipalData::from(StacksAddress::burn_address(false)) ; "standard address")]
     #[test_case(PrincipalData::parse(CONTRACT_ADDRESS).unwrap(); "contract address")]
-    fn btc_address(stacks_address: PrincipalData) {
+    fn btc_address(recipient: PrincipalData) {
```
Comment (AshtonStephens) [inline common/src/deposits.rs:254]: 👍

**Review by AshtonStephens — APPROVED** [review]
This looks good to me. I'm still not clear on how we plan to verify the reclaim script from https://github.com/stacks-network/sbtc/issues/30. Is that going to be a separate issue?

```
<lock_time> CHECKSEQUENCEVERIFY <reclaim script>
```

---
*Edit:* This will be addressed in a later ticket: https://github.com/stacks-network/sbtc/pull/333#discussion_r1678300773

File: common/src/deposits.rs [hunk 2]
```diff
@@ -83,11 +116,79 @@ impl DepositScript {
...
+/// ```
+pub fn parse_deposit_script(deposit_script: &ScriptBuf) -> Result<DepositScriptInputs, Error> {
+    let script = deposit_script.as_bytes();
+
+    // Valid deposit scripts cannot be less than this length.
+    if script.len() < STANDARD_SCRIPT_LENGTH {
+        return Err(Error::InvalidDepositScript);
+    }
+    // This cannot panic because of the above check and the fact that
+    // DEPOSIT_SCRIPT_FIXED_LENGTH < STANDARD_SCRIPT_LENGTH.
```
Comment (netrome) [inline common/src/deposits.rs:143]: Oh I didn't realize `split_at` was a panicking function. Thanks for the clarification!

**Review by netrome — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**